libmuscle/python/libmuscle/manager/qcgpj_instantiator.py

```diff
@@ -178,7 +182,8 @@ async def _main(self) -> None:
                 except queue.Empty:
                     break
 
-            await asyncio.sleep(0.1)
```

Comment:
This code is a bit complicated, and I needed to read it again, but it seems to me that we can just move the sleep statement into an `if not done:` block at the end of the outer loop? That way we first drain the requests queue and submit all the launch and cancel requests to QCG-PJ, and then we process all finished processes right away, and then we check whether we should quit and if not, sleep. And we avoid an extra variable. What do you think?